djangocms_text_ckeditor/static/djangocms_text_ckeditor/ckeditor_plugins/cmsplugins/plugin.js

```diff
@@ -1,14 +1,36 @@
-(function($) {
-// CMS.$ will be passed for $
-$(document).ready(function () {
+(function ($) {
+    /**
+     * Determine if we should return `div` or `span` based on the
+     * plugin markup.
+     *
+     * @function getFakePluginElement
+     * @private
+     * @param {String} pluginMarkup valid html hopefully
+     * @returns {String} div|span
+     */
+    function getFakePluginElement(pluginMarkup) {
+        var innerTags = (pluginMarkup.match(/<([\S]*?)\s[\s\S]*?>/) || [0, false]).splice(1);
+
+        var containsAnyBlockLikeElements = innerTags.some(function (tag) {
+            return tag && CKEDITOR.dtd.$block[tag];
+        });
+
+        var fakeRealType = 'span';
+
+        if (containsAnyBlockLikeElements) {
+            fakeRealType = 'div';
+        }
+
+        return fakeRealType;
```

Comment:
would prefer to keep longer version, no real difference in understanding, at some point uglify will convert it to ternary automatically